pkg/rescheduler/strategies/duplicates_test.go

```diff
@@ -0,0 +1,166 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	core "k8s.io/client-go/testing"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/clientset/fake"
+)
+
+// TODO:@ravisantoshgudimetla. As of now building some test pods here. This needs to
+// move to utils after refactor.
+// buildTestPod creates a test pod with given parameters.
+func buildTestPod(name string, cpu int64, memory int64, nodeName string) *v1.Pod {
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "default",
+			Name:      name,
+			SelfLink:  fmt.Sprintf("/api/v1/namespaces/default/pods/%s", name),
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Resources: v1.ResourceRequirements{
+						Requests: v1.ResourceList{},
+					},
+				},
+			},
+			NodeName: nodeName,
+		},
+	}
+	if cpu >= 0 {
+		pod.Spec.Containers[0].Resources.Requests[v1.ResourceCPU] = *resource.NewMilliQuantity(cpu, resource.DecimalSI)
+	}
+	if memory >= 0 {
+		pod.Spec.Containers[0].Resources.Requests[v1.ResourceMemory] = *resource.NewQuantity(memory, resource.DecimalSI)
+	}
+
+	return pod
+}
+
+// buildTestNode creates a node with specified capacity.
+func buildTestNode(name string, millicpu int64, mem int64) *v1.Node {
+	node := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:     name,
+			SelfLink: fmt.Sprintf("/api/v1/nodes/%s", name),
+			Labels:   map[string]string{},
+		},
+		Status: v1.NodeStatus{
+			Capacity: v1.ResourceList{
+				v1.ResourcePods:   *resource.NewQuantity(100, resource.DecimalSI),
+				v1.ResourceCPU:    *resource.NewMilliQuantity(millicpu, resource.DecimalSI),
+				v1.ResourceMemory: *resource.NewQuantity(mem, resource.DecimalSI),
+			},
+			Allocatable: v1.ResourceList{
+				v1.ResourcePods:   *resource.NewQuantity(100, resource.DecimalSI),
+				v1.ResourceCPU:    *resource.NewMilliQuantity(millicpu, resource.DecimalSI),
+				v1.ResourceMemory: *resource.NewQuantity(mem, resource.DecimalSI),
+			},
+			Phase: v1.NodeRunning,
+			Conditions: []v1.NodeCondition{
+				{Type: v1.NodeReady, Status: v1.ConditionTrue},
+			},
+		},
+	}
+	return node
+}
+
+// getMirrorPodAnnotation returns the annotation needed for mirror pod.
+func getMirrorPodAnnotation() map[string]string {
+	return map[string]string{
+		"kubernetes.io/created-by":    "{\"kind\":\"SerializedReference\",\"apiVersion\":\"v1\",\"reference\":{\"kind\":\"Pod\"}}",
+		"kubernetes.io/config.source": "api",
+		"kubernetes.io/config.mirror": "mirror",
+	}
+}
+
+// getNormalPodAnnotation returns the annotation needed for a normal pod. A normal pod is one without any references to
+func getNormalPodAnnotation() map[string]string {
+	return map[string]string{
+		"kubernetes.io/created-by": "{\"kind\":\"SerializedReference\",\"apiVersion\":\"v1\",\"reference\":{\"kind\":\"Pod\"}}",
+	}
+}
+
+// getReplicaSetAnnotation returns the annotation needed for replicaset pod.
+func getReplicaSetAnnotation() map[string]string {
```

Comment:
Its ok for time being. Later you could add for RS, Deployment and Job too.

Reply:
Sure, I am focusing on code coverage as of now.